packages/template-retail-react-app/app/pages/social-login-redirect/index.jsx

```diff
@@ -14,19 +14,64 @@ import {
     Text,
     Spinner
 } from '@salesforce/retail-react-app/app/components/shared/ui'
-import {useCustomerType} from '@salesforce/commerce-sdk-react'
+
+// Hooks
 import useNavigation from '@salesforce/retail-react-app/app/hooks/use-navigation'
+import {useAuthHelper, AuthHelpers, useShopperBasketsMutation} from '@salesforce/commerce-sdk-react'
+import {useSearchParams} from '@salesforce/retail-react-app/app/hooks'
+import {useCurrentCustomer} from '@salesforce/retail-react-app/app/hooks/use-current-customer'
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+import {useAppOrigin} from '@salesforce/retail-react-app/app/hooks/use-app-origin'
+import {
+    getSessionJSONItem,
+    clearSessionJSONItem,
+    buildRedirectURI
+} from '@salesforce/retail-react-app/app/utils/utils'
 
 const SocialLoginRedirect = () => {
     const navigate = useNavigation()
-    const {isRegistered} = useCustomerType()
+    const [searchParams] = useSearchParams()
+    const loginIDPUser = useAuthHelper(AuthHelpers.LoginIDPUser)
+    const {data: customer} = useCurrentCustomer()
+    // Build redirectURI from config values
+    const appOrigin = useAppOrigin()
+    const redirectPath = getConfig().app.login.social?.redirectURI || ''
+    const redirectURI = buildRedirectURI(appOrigin, redirectPath)
+
+    const locatedFrom = getSessionJSONItem('returnToPage')
+    const mergeBasket = useShopperBasketsMutation('mergeBasket')
+
+    // Runs after successful 3rd-party IDP authorization, processing query parameters
+    useEffect(() => {
+        if (searchParams.code && searchParams.usid) {
+            loginIDPUser.mutateAsync({
+                code: searchParams.code,
+                redirectURI: redirectURI
+            })
+        }
+    }, [])
 
-    // If customer is registered push to account page
+    // If customer is registered, push to secure account page
     useEffect(() => {
-        if (isRegistered) {
-            navigate('/account')
+        if (customer?.isRegistered) {
+            clearSessionJSONItem('returnToPage')
+            mergeBasket.mutate({
+                headers: {
+                    // This is not required since the request has no body
+                    // but CommerceAPI throws a '419 - Unsupported Media Type' error if this header is removed.
+                    'Content-Type': 'application/json'
+                },
```

Comment:
I noticed that this is a copy of out other basket merge. I'm wondering if the commerce-sdk-react should be adding this for us. Kevin might be a good person to ask about that. Looking at the api docs the body is alway json, so I don't know why we would have to specify that all the time.